.github/workflows/linux_qemu.yml

```diff
@@ -175,3 +175,86 @@ jobs:
           '"
 
 
+  linux_loongarch64_qemu:
```

Comment:
This job takes 46 minutes, which is not acceptable. We need to speed up the build by using a native toolchain similar to the other architectures. There are two options here:

- Since LoongArch requires Ubuntu 24.04, you can try upgrading the main matrix OS from 22.04 to 24.04 and add LoongArch within it.
- If the option above doesn't work, then you will need to reuse the same toolchain. In this case, the host must share x86_64 binaries with the container. Binaries for various architectures can coexist within a single container, and the binfmt hook primarily channels exotic binaries to QEMU while bypassing native calls. see https://github.com/numpy/numpy/blob/9e76ce7a30db4dfef0884203b94c1fb6d2994100/.github/workflows/linux_qemu.yml#L132-L146

Reply:
I uploaded an image to Docker Hub to avoid the process of downloading the necessary software inside the container.

Reply:
I created a self-hosted runner (Linux, x86， ubuntu) and conducted a test. The test results are as follows: https://github.com/CNClareChen/numpy-pengxu/actions/runs/12252270750/job/3417860558. May I ask if there are any similar runners available in this repository for me to choose from?

Reply:
I created a self-hosted runner (Linux, x86， ubuntu) and conducted a test. The test results are as follows: https://github.com/CNClareChen/numpy-pengxu/actions/runs/12252270750/job/3417860558. May I ask if there are any similar runners available in this repository for me to choose from?

Reply:
The problem is not the runner image, since any qemu emulation of loongarch is going to be slow. We need to build outside the image via cross-compilation. Also CI is failing. Maybe go back to the last good commit.

Reply:
The reason for the compilation failure is due to the commit 099c44e452ad1fa1643d8012ed6871e3139f8066, which updated highway from commit a97b5d371d696564e206627a883b1341c65bd983 to commit 68b0fdebffb14f3b8473fed1c33ce368efc431e7.

Reply:
After communicating with the Highway community, we implemented a workaround for compiling NumPy on the LoongArch platform by temporarily adding the option CXXFLAGS="-DHWY_COMPILE_ONLY_SCALAR". Currently, the CI tests have been successfully passed